src/Views/Opportunity/QuickEdit.js

```diff
@@ -0,0 +1,150 @@
+    string,
+    validator,
+    template,
+    utility,
+    salesProcessUtility,
+    Edit
+) {
+
+    var __class = declare('crm.Views.Opportunity.QuickEdit', [Edit], {
+        //Localization
+        estCloseText: 'est close',
+        detailsText: 'Details',
+        opportunityStageTitleText: 'Opportunity Stage',
+        opportunityText: 'opportunity',
+        stageText: 'stage',
+        salesProcessText: 'locked by sales process',
+
+        //View Properties
+        entityName: 'Opportunity',
+        id: 'opportunity_quick_edit',
+        resourceKind: 'opportunities',
+        insertSecurity: 'Entities/Opportunity/Add',
+        updateSecurity: 'Entities/Opportunity/Edit',
+        querySelect: [
+            'Account/AccountName',
+            'EstimatedClose',
+            'Stage'
+        ],
+        init: function() {
+            this.inherited(arguments);
+        },
+        applyContext: function(templateEntry) {
+            this.fields['EstimatedClose'].setValue(templateEntry.EstimatedClose);
+        },
+        createLayout: function() {
+            var layout, details;
+
+            details = {
+                title: this.detailsText,
+                name: 'OpportunityDetailsEdit',
+                children: [{
+                    relatedView: {
+                        widgetType: 'relatedContext',
+                        id: 'opp_related_context_quickEdit'
+                    }
+                },
+                    {
+                        label: this.stageText,
+                        name: 'Stage',
+                        property: 'Stage',
+                        picklist: 'Opportunity Stage',
+                        requireSelection: true,
+                        enabled: false,
+                        title: this.opportunityStageTitleText,
+                        type: 'picklist'
+                    },
+                    {
+                        label: this.estCloseText,
+                        name: 'EstimatedClose',
+                        property: 'EstimatedClose',
+                        type: 'date',
+                        timeless: true,
+                        validator: validator.exists
+                    }
+                ]
+            };
+
+            layout = this.layout || (this.layout = []);
+
+            if (layout.length > 0) {
+                return layout;
+            }
+
+            layout.push(details);
+            return layout;
+        },
+        setValues: function(values) {
+            this.enableStage(values['$key']);
+            this.inherited(arguments);
+
+        },
+        enableStage: function(opportunityId) {
+            var promise, field, label;
+            field = this.fields['Stage'];
+            label = this.stageText;
+            if (!field) {
+                return;
+            }
+            field.disable();
+            promise = salesProcessUtility.getSalesProcessByEntityId(opportunityId);
+            if (promise) {
```

Comment:
getSalesProcessByEntityId always returns a promise object, which will always be truthy, no reason to check it.